passes/techmap/dffinit.cc

```diff
@@ -135,10 +135,18 @@ struct DffinitPass : public Pass {
 							continue;
 						while (GetSize(value.bits) <= i)
 							value.bits.push_back(State::S0);
-						if (noreinit && value.bits[i] != State::Sx && value.bits[i] != init_bits.at(sig[i]))
-							log_error("Trying to assign a different init value for %s.%s.%s which technically "
-									"have a conflicted init value.\n",
-									log_id(module), log_id(cell), log_id(it.second));
+						if (noreinit && value.bits[i] != State::Sx && value.bits[i] != init_bits.at(sig[i])) {
+							if (init_bits.at(sig[i]) != State::Sx) {
+								log_error("Trying to assign a different init value for %s.%s.%s which technically "
+										"have a conflicted init value.\n",
+										log_id(module), log_id(cell), log_id(it.second));
+							}
+							else {
+								// Trying to overwrite an existing INIT value with 1'bx,
+								//   silently ignore?
+								continue;
```

Comment:
More consistent with other passes would be to add `if (value[i] != State::Sx)` in the code quoted below so that X values are not even added to `init_bits`. (I think the proper behavior would be to not act on X bits in init attributes at all, independent of `-noreinit`.)

https://github.com/YosysHQ/yosys/blob/1f62dc9081feb4852b1848d01951f631853edb38/passes/techmap/dffinit.cc#L103-L105